### PR TITLE
[SPARK-36070][CORE] Log time cost info for writing rows out and committing the task

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

We have a job that has a stage that contains about 8k tasks.  Most tasks take about 1~10min to finish but 3 of them tasks run extremely slow with similar data sizes. They take about 1 hour each to finish and also do their speculations. 

The root cause is most likely the delay of the storage system. But it's not straightforward enough to find where the performance issue occurs, in the phase of shuffle read, task execution, output, commitment e.t.c..

```log
2021-07-09 03:05:17 CST SparkHadoopMapRedUtil INFO - attempt_20210709022249_0003_m_007050_37351: Committed
2021-07-09 03:05:17 CST Executor INFO - Finished task 7050.0 in stage 3.0 (TID 37351). 3311 bytes result sent to driver
2021-07-09 04:06:10 CST ShuffleBlockFetcherIterator INFO - Getting 9 non-empty blocks including 0 local blocks and 9 remote blocks
2021-07-09 04:06:10 CST TransportClientFactory INFO - Found inactive connection to
```

### Why are the changes needed?

On the spark side, we can record the time cost in logs for better bug hunting or performance tuning.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

passing GA

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu @dongjoon-hyun, thanks

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala [hunk 1]
```diff
@@ -301,8 +301,14 @@ object FileFormatWriter extends Logging {
...
         // Execute the task to write rows out and commit the task.
-        dataWriter.writeWithIterator(iterator)
-        dataWriter.commit()
+        val taskAttemptID = taskAttemptContext.getTaskAttemptID
+        val (res, timeCost) = Utils.timeTakenMs {
+          logDebug("$taskAttemptID starts to write and commit.")
+          dataWriter.writeWithIterator(iterator)
+          dataWriter.commit()
+        }
+        logInfo(s"$taskAttemptID finished to write and commit. Elapsed time: $timeCost ms.")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:310]: After some more thought, I think it's better to use SQL metrics for it. It's very hard to know max/min/avg by reading the logs.

@AngersZhuuuu I think you tried it before. Can you restore the work?

Comment (AngersZhuuuu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala:310]: > After some more thought, I think it's better to use SQL metrics for it. It's very hard to know max/min/avg by reading the logs.
> 
> @AngersZhuuuu I think you tried it before. Can you restore the work?

Yea, working on this

**Pull request closed without merging**